src/java.base/share/classes/java/util/zip/ZipEntry.java

```diff
@@ -37,6 +37,11 @@
 
 /**
  * This class is used to represent a ZIP file entry.
+ * <P>
+ * The combined length of the entry name, the extra field data, the
+ * entry comment and {@linkplain #CENHDR CEN Header size}, must not
+ * exceed 65,535 bytes. If it does, {@linkplain ZipOutputStream} will
+ * throw a {@linkplain ZipException} when writing the ZIP file entry.
```

Comment:
This looks a little out of place in ZipEntry's class description, does ZOS.putNextEntry throw or is it just finish and close?

Reply:
Short answer. finish() which calls writeCEN, will throw for the above.

 As the entry comment, is only part of the CEN, I wanted to keep the encoding in writeCEN as there is no reason to do it earlier.

Reply:
I looks very out of place when reading ZipEntry's class description. I think we'll have to move to the places where the exception is thrown.

Reply:
So that means removing completely from ZipEntry, which is fine.  The wording is in ZipOutputStream::finish() and was in ZipOutputStream::close(), but I believe with your proposed change, we are removing it from close